engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -15,23 +15,25 @@
  */
 package org.terasology.rendering.dag;
 
+import com.google.api.client.util.Lists;
+import java.util.List;
+
 /**
- *
+ * TODO: Add javadocs
  */
-public class RenderPipelineGenerator {
-    private RenderGraph renderGraph;
+public final class RenderTaskListGenerator {
+
+    private RenderTaskListGenerator() {
 
-    public RenderPipelineGenerator(RenderGraph renderGraph) {
-        this.renderGraph = renderGraph;
     }
 
-    public RenderPipeline generate() {
-        RenderPipeline renderPipeline = new RenderPipeline();
-        for (Node node : renderGraph.getNodesInTopologicalOrder()) {
+    public static List<RenderPipelineTask> createFrom(List<Node> orderedNodes) {
+        List<Object> renderPipeline = Lists.newArrayList();
+        for (Node node : orderedNodes) {
             // TODO: add "renderPipeline.addAll(node.getDesiredStateChanges())" here
             renderPipeline.add(node);
         }
-
-        return renderPipeline;
+        RenderPipelineOptimizer optimizer = new RenderPipelineOptimizer(renderPipeline);
```

Comment:
In line with what written ago, I'd simply add the optimizing code to this class rather than its own separate class.

The optimizer made sense if we had a RenderPipelineGenerator and a RenderPipelineOptimizer. But if the intermediate step of generating the unoptimized pipeline is handled internally I think we can skip the optimizer class and have its functionality here, in the RenderTaskListGenerator.